extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/accesslog/AccessLogHandler.java

```diff
@@ -152,10 +157,25 @@ public void handle(RoutingContext rc) {
             rc.next();
             return;
         }
-        QuarkusRequestWrapper.get(rc.request()).addRequestDoneHandler(new Handler<Void>() {
-            @Override
-            public void handle(Void event) {
-                accessLogReceiver.logMessage(tokens.readAttribute(rc));
+        QuarkusRequestWrapper.get(rc.request()).addRequestDoneHandler(event -> {
+            Map<String, Optional<String>> serialized = ((ExchangeAttributeSerializable) tokens).serialize(rc);
+            Map<String, Object> entries = new HashMap<>();
+
+            for (Map.Entry<String, Optional<String>> entry : serialized.entrySet()) {
+                if (entry.getValue().isPresent()) {
+                    entries.put(entry.getKey(), MDC.get(entry.getKey()));
+                    MDC.put(entry.getKey(), entry.getValue().get());
+                }
+            }
+
+            accessLogReceiver.logMessage(tokens.readAttribute(rc));
+
+            for (Map.Entry<String, Object> entry : entries.entrySet()) {
+                if (entry.getValue() != null) {
+                    MDC.put(entry.getKey(), entry.getValue());
+                } else {
+                    MDC.remove(entry.getKey());
+                }
```

Comment:
Are we sure all the people using the access log will want to pay this price? I foresee quite a bunch of allocations to read the attribute and add them to the Map. I'm not convinced it makes sense to make it the default behavior from a performance perspective.